url594c007908_hugobraga_College_tgz-pJ8-main_GraphJ8/src/dataStructure/AdjMatrix.java

```diff
@@ -0,0 +1,455 @@
+    		}
+    		
+    		
+    		
+    		if (cTerminals > sqrtT) {//here, sqrtT must contain the original value
+    			if (terminalRoot) {
+    				cTerminals--;
+        			for (int t = 0; t < terminalsIndex.length; t++) {
+        				if (terminalsIndex[t] == rootIndex) {
+        					terminalsIndex[t] = -1;
+        					break;
+        				}
+        			}
+        			local2CheckAdjList.deleteNode(rootIndex);
+    			}
+    			Arrays.sort(pairDelays, 0, (cTerminals), new S2DDelayComparator());
+
+    			TuAdjl = new AdjList(size);
+        		//insert the paths in Tu
+    			int closestTerQt = (int)Math.floor(sqrtT);
+    			if (terminalRoot)
+    				closestTerQt--;
+    			for (int c = 0; c < closestTerQt; c++) {
+    				int prev = pairDelays[c].getDestination();
+    				for (int t = 0; t < terminalsIndex.length; t++) {
+    					if (terminalsIndex[t] == prev) {
+    						terminalsIndex[t] = -1;
+    						break;
+    					}
+    				}
+        			while (P[rootIndex][prev] != rootIndex) {
+        				TuAdjl.checkInsertEdge(P[rootIndex][prev], prev, MCPmatrix[P[rootIndex][prev]][prev]);
+        				local2CheckAdjList.deleteNode(prev);
+        				prev = P[rootIndex][prev];
+        			}
+
+        			TuAdjl.checkInsertEdge(P[rootIndex][prev], prev, MCPmatrix[P[rootIndex][prev]][prev]);
+        			local2CheckAdjList.deleteNode(prev);
+    			}
+    			local2CheckAdjList.deleteNode(rootIndex);
+    			root[0] = rootIndex;
+    			lastPossibleRoot = i;
+    			fillMatrixFromAdjList(local2CheckAdjList);
+    			minCostPaths();
+    			localAdjList = local2CheckAdjList;
+    			return TuAdjl;
+    		}
+    	}   
+    	lastPossibleRoot = -1;
+    	localAdjList = local2CheckAdjList;
+    	return null;
+    }
+    
+
+    public ALNode getReducedNeibrsFromMCPMatrix (int index) { 
+    	ALNode adjl = null;
+    	
+    	for (int j = 0; j < size; j++) {
+    	    if (P[index][j] == index) {
+    		adjl = insertInList(adjl, j, MCPmatrix[index][j]);
+    	    }
+    	}
+    	
+	return adjl;
+    }
+    
+    public void printSP(int source, int d) {
+		int prev = d;
+		double sumCost = 0;
+		Stack<Integer> stack = new Stack<Integer>();
+		if (source != d) {
+			stack.push(prev);
+			while (P[source][prev] != source) {
+				stack.push(P[source][prev]);
+				sumCost += MCPmatrix[P[source][prev]][prev];
+				prev = P[source][prev];
+			}
+			stack.push(P[source][prev]);
+			sumCost += MCPmatrix[P[source][prev]][prev];
+		}
+		//System.out.println("----------");
+		System.out.println("custo somado de "+source+" para "+d+": "+sumCost+", pela matrix: "+MCPmatrix[source][d]);
+		System.out.print("caminho de "+source+" para "+d+": ");
+		while (!stack.isEmpty()) {
+			System.out.print(stack.pop()+", ");
+		}
+		System.out.println("");
+		//System.out.println("----------");
+		
+		
+    }
+
+
+    // Methods related to the implementation of Floyd-Warshall's Minimum Cost Paths algorithm
+
+    public void minCostPaths () {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.